packages/db/src/search/CollectionSearch.ts

```diff
@@ -66,21 +92,85 @@ export class CollectionSearch extends BaseSearch {
+  protected translateAllPostgresRecordsToApiRecords(pgRecords: QueryCollectionRecord[],
+    stats?: IncludeStatsRecord): Partial<CollectionRecordApi>[] {
+    const apiRecords = pgRecords.map((record) => {
+      const apiRecord: CollectionRecordApi = translatePostgresCollectionToApiCollection(record);
+      const apiRecordFinal = this.dbQueryParameters.fields
         ? pick(apiRecord, this.dbQueryParameters.fields)
         : apiRecord;
+
+      if (stats) {
+        apiRecordFinal.stats = stats[record.cumulus_id] ? stats[record.cumulus_id].statuses : {
+          queued: 0,
+          completed: 0,
+          failed: 0,
+          running: 0,
+          total: 0,
+        };
+      }
+      return apiRecordFinal;
     });
     return apiRecords;
   }
+
+  /**
+   * Executes stats query to get granules' stats info for collections if includeStats is true
+   *
+   * @param ids - array of cumulusIds of the collections
+   * @param knex - knex for the stats query
+   * @returns the collection's granules status' aggregation
+   */
+  protected async executeStatsQuery(ids: number[], knex: Knex): Promise<IncludeStatsRecord> {
+    const granulesTable = TableNames.granules;
+    const statsQuery = knex(granulesTable)
+      .select(`${granulesTable}.collection_cumulus_id`, `${granulesTable}.status`)
+      .count(`${granulesTable}.status`)
+      .groupBy(`${granulesTable}.collection_cumulus_id`, `${granulesTable}.status`)
+      .whereIn(`${granulesTable}.collection_cumulus_id`, ids);
+    const results = await statsQuery;
+    const reduced = results.reduce((acc, record) => {
+      const cumulusId = record.collection_cumulus_id;
+      if (!acc[cumulusId]) {
+        acc[cumulusId] = {
+          id: cumulusId,
+          statuses: {
+            queued: 0,
+            completed: 0,
+            failed: 0,
+            running: 0,
+            total: 0,
+          },
+        };
+      }
+      acc[cumulusId].statuses[record.status as keyof Statuses] += Number(record.count);
+      acc[cumulusId].statuses['total'] += Number(record.count);
+      return acc;
+    }, {} as IncludeStatsRecord);
+    return reduced;
+  }
+
+  /**
+   * Translate postgres records to api records
+   *
+   * @param pgRecords - postgres records returned from query
+   * @param knex - knex for the stats query if incldueStats is true
+   * @returns translated api records
+   */
+  protected async translatePostgresRecordsToApiRecords(pgRecords: QueryCollectionRecord[],
+    knex: Knex): Promise<Partial<CollectionRecordApi>[]> {
+    log.debug(`translatePostgresRecordsToApiRecords number of records ${pgRecords.length} `);
+    if (this.includeStats) {
+      const cumulusIds = pgRecords.map((record) => record.cumulus_id);
+      const reduced = await this.executeStatsQuery(cumulusIds, knex);
+      return this.translateAllPostgresRecordsToApiRecords(pgRecords, reduced);
+    }
+    return this.translateAllPostgresRecordsToApiRecords(pgRecords);
```

Comment:
Feel like no need to have a separate translateAllPostgresRecordsToApiRecords method